packages/llvm.rb

```diff
@@ -285,51 +47,44 @@ def self.build
 cxx_inc=#{CREW_PREFIX}/include/c++/\${version}/\${machine}
 gnuc_lib=#{CREW_LIB_PREFIX}/gcc/\${machine}/\${version}
 clang++ -fPIC  -rtlib=compiler-rt -stdlib=libc++ -cxx-isystem \${cxx_sys} -I \${cxx_inc} -B \${gnuc_lib} -L \${gnuc_lib} \"\$@\"' > clc++"
-      case ARCH
-      when "x86_64"
-        system 'cmake',
-               "-DCURSES_INCLUDE_PATH='#{CREW_PREFIX}/include/ncursesw'",
-               "-DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX}",
-               '-DCMAKE_CXX_FLAGS="-fPIC"',
-               '-DCMAKE_BUILD_TYPE=Release',
-               '-DLLVM_LIBDIR_SUFFIX=64',
-               '-DBUILD_SHARED_LIBS=ON',
-               '-DLLVM_ENABLE_RTTI=ON',
-               '-Wno-dev',
-               '..'
-               # Fix for file INSTALL cannot find ".../lib64/python2.7" error.
-               # See http://lists.llvm.org/pipermail/lldb-dev/2015-June/007633.html.
-               system "sed -i '40,43d' tools/lldb/cmake_install.cmake"
-      else # armv7l, aarch64 or i686
-        system 'cmake',
-               "-DCURSES_INCLUDE_PATH='#{CREW_PREFIX}/include/ncursesw'",
-               "-DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX}",
-               '-DCMAKE_CXX_FLAGS="-fPIC"',
-               '-DCMAKE_BUILD_TYPE=Release',
-               '-DBUILD_SHARED_LIBS=ON',
-               '-DLLVM_ENABLE_RTTI=ON',
-               '-Wno-dev',
-               '..'
-      end
-      system 'make'
+
+        system "cmake -G Ninja \
+            -DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX} \
+            -DLLVM_DEFAULT_TARGET_TRIPLE=#{ARCH}-cros-linux-gnu \
+            -DLLVM_TARGETS_TO_BUILD=#{LLVM_TARGETS_TO_BUILD} \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DLLVM_LIBDIR_SUFFIX='#{CREW_LIB_SUFFIX}' \
+            -DLLVM_BUILD_LLVM_DYLIB=ON \
+            -DLLVM_ENABLE_RTTI=ON \
+            -DLLVM_USE_LINKER=gold \
+            -DLLVM_INSTALL_UTILS=ON \
+            -DLLVM_BINUTILS_INCDIR='../binutils/include' \
+            -DLLVM_ENABLE_PROJECTS='clang;clang-tools-extra;libcxx;libcxxabi;libunwind;lldb;compiler-rt;lld;polly;openmp' \
+            -Wno-dev \
+            ../llvm"
+      system 'ninja'
     end
-    system "ld_default #{original_default}"
   end
 
   def self.install
     Dir.chdir("builddir") do
       system "install -Dm755 clc #{CREW_DEST_PREFIX}/bin/clc"
       system "install -Dm755 clc++ #{CREW_DEST_PREFIX}/bin/clc++"
-      system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
+      system "DESTDIR=#{CREW_DEST_DIR} ninja install"
+      FileUtils.mkdir_p "#{CREW_DEST_LIB_PREFIX}/bfd-plugins"
+      #FileUtils.cp lib#{CREW_LIB_SUFFIX}/LLVMgold.so "#{CREW_DEST_LIB_PREFIX}/bfd-plugins/"
+      #FileUtils.cp lib#{CREW_LIB_SUFFIX}/libLTO.* "#{CREW_DEST_LIB_PREFIX}/bfd-plugins/"
+      system "cp lib#{CREW_LIB_SUFFIX}/LLVMgold.so #{CREW_DEST_LIB_PREFIX}/bfd-plugins/"
+      system "cp lib#{CREW_LIB_SUFFIX}/libLTO.* #{CREW_DEST_LIB_PREFIX}/bfd-plugins/"
```

Comment:
Rubyize?
```
FileUtils.cp "lib#{CREW_LIB_SUFFIX}/LLVMgold.so", "#{CREW_DEST_LIB_PREFIX}/bfd-plugins/"
FileUtils.cp Dir.glob("lib#{CREW_LIB_SUFFIX}/libLTO.*"), "#{CREW_DEST_LIB_PREFIX}/bfd-plugins/"
```

Reply:
That's exactly the help I was looking for so as not to mess that up.